core/src/main/java/org/apache/iceberg/rest/requests/UpdateTableRequest.java

```diff
@@ -42,13 +46,19 @@ public UpdateTableRequest() {
     // needed for Jackson deserialization
   }
 
+  /**
+   * @deprecated will be removed in 1.5.0; use {@link UpdateTableRequestBuilder#newBuilder()}
+   *     instead.
+   */
+  @Deprecated
   public UpdateTableRequest(List<UpdateRequirement> requirements, List<MetadataUpdate> updates) {
     this.requirements = requirements;
     this.updates = updates;
   }
 
+  @Constructor
```

Comment:
this will generate a Builder class from the constructor params and we're indicating in the constructor that `TableIdentifier` can be null.

Reply:
the constructor needs to be non-private (we could also change it to `protected` to be more strict and force usage via the generated builder